src/parsers/timeseries_formats.jl

```diff
@@ -0,0 +1,234 @@
+        # TODO DT: Hack. Do we really want to support this?
+        # If so, can we check for a specific name?  "VALUE" is used in the examples.
+        @warn "column name doesn't match...resetting" value_columns component_name
+        value_columns[1] = Symbol(component_name)
+    end
+
+    return TimeSeries.TimeArray(timestamps, hcat(vals...), value_columns)
+end
+
+"""This version of the function supports the format where there is no column header for
+a component, so the component_name must be passed in.
+"""
+function read_time_array(
+                         ::Type{T},
+                         file::CSV.File,
+                         component_name::AbstractString;
+                         kwargs...
+                        ) where T <: TimeseriesFormatPeriodAsHeader
+    if length(file) != 1
+        msg = "$T must have only one row. file=$file.name"
+        throw(DataFormatError(msg))
+    end
+
+    timestamps = Vector{Dates.DateTime}()
+
+    period_cols_as_symbols = get_period_columns(T, file)
+    period = [parse(Int, string(x)) for x in period_cols_as_symbols]
+    step = get_step_time(T, file, period)
+
+    for i in 1:length(period)
+        timestamp = get_timestamp(T, file, 1) + step * (i - 1)
+        push!(timestamps, timestamp)
+    end
+
+    vals = [getproperty(file, x)[1] for x in period_cols_as_symbols]
+
+    return TimeSeries.TimeArray(timestamps, vals, Symbol.([component_name]))
+end
+
+"""This version of the function only has component_name to match the interface.
+It is unused and shouldn't be passed.
+
+Set start_datetime as a keyword argument for the starting timestamp, otherwise the current
+day is used.
+"""
+function read_time_array(
+                         ::Type{T},
+                         file::CSV.File,
+                         component_name=nothing;
+                         kwargs...
+                        ) where T <: TimeseriesFormatComponentsAsColumnsNoTime
+    @assert isnothing(component_name)
+
+    timestamps = Vector{Dates.DateTime}()
+    step = get_step_time(T, file)
+
+    start = get(kwargs, :start_datetime, Dates.DateTime(Dates.today()))
+    for i in 1:length(file)
+        timestamp = start + step * (i - 1)
+        push!(timestamps, timestamp)
+    end
+
+    value_columns = get_value_columns(T, file)
+    vals = [getproperty(file, x) for x in value_columns]
+
+    return TimeSeries.TimeArray(timestamps, hcat(vals...), value_columns)
+end
+
+"""Return a DateTime for the step between values as specified by the period in the file."""
+function get_step_time(
+                       ::Type{T},
+                       file::CSV.File,
+                       period::AbstractArray,
+                      ) where T <: TimeseriesFormat
+    @assert period[end] == maximum(period) == length(period)
+    num_steps = period[end]
+
+    timestamps = get_unique_timestamps(T, file)
+    if length(timestamps) == 1
```

Comment:
What do you all think about this?

Reply:
for `DateTime` with a single value, we should assume 1 hour resolution.